kpops/utils/gen_schema.py

```diff
@@ -76,32 +76,65 @@ def is_valid_component(
         return True
 
 
-def gen_pipeline_schema(
-    components_module: str | None = None,
-) -> None:
+def _add_components(
+    components_module: str, components: tuple[type[PipelineComponent]] | None = None
+) -> tuple[type[PipelineComponent]]:
+    """Add components to a components tuple
+
+    If an empty tuple is provided or it is not provided at all, the components
+    types from the given module are 'tupled'
+
+    :param components_module: Python module. Only the classes that inherit from
+        PipelineComponent will be considered.
+    :type components_module: str
+    :param components: Tuple of components to which to add, defaults to ()
+    :type components: tuple, optional
+    :return: Extended tuple
+    :rtype: tuple
     """
-    Generate a json schema from the models of pipeline components.
+    if components is None:
+        components = tuple()  # type: ignore[assignment]
+    # HACK: mypy doesn't narrow the tuple's type without this assertion
+    assert components is not None
```

Comment:
I guess this is not needed anymore then?

Reply:
This removes 3 more errors haha

Here is what happens without it and with the 3 silencing comments:

```
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

kpops/utils/gen_schema.py:102: error: Item "None" of
"Optional[Tuple[Type[PipelineComponent]]]" has no attribute "__iter__" (not
iterable)  [union-attr]
            for component in components
                             ^~~~~~~~~~
kpops/utils/gen_schema.py:110: error: Unsupported left operand type for +
("None")  [operator]
        components += tuple(custom_components)  # type: ignore[assignment]
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
kpops/utils/gen_schema.py:110: note: Error code "operator" not covered by "type: ignore" comment
kpops/utils/gen_schema.py:110: note: Left operand is of type "Optional[Tuple[Type[PipelineComponent]]]"
kpops/utils/gen_schema.py:111: error: Incompatible return value type (got
"Optional[Tuple[Type[PipelineComponent]]]", expected
"Tuple[Type[PipelineComponent]]")  [return-value]
        return components
               ^~~~~~~~~~
Found 3 errors in 1 file (checked 109 source files)
```